lazy_loader/__init__.py

```diff
@@ -160,38 +166,72 @@ def myfunc():
         Actual loading of the module occurs upon first attribute request.
 
     """
-    try:
-        return sys.modules[fullname]
-    except KeyError:
-        pass
+    module = sys.modules.get(fullname)
+    have_module = module is not None
+
+    # Most common, short-circuit
+    if have_module and require is None:
+        return module
+
+    spec = None
+    if not have_module:
+        spec = importlib.util.find_spec(fullname)
+        have_module = spec is not None
+
+    if not have_module:
+        not_found_message = f"No module named '{fullname}'"
+    elif require is not None:
+        # Old style lazy loading to avoid polluting sys.modules
```

Comment:
What does this comment mean?

Reply:
Ah. Initially I'd used `load()` to lazily load `packaging.requirements` at the module root, but it seems better not to add it to `sys.modules` unless calling code actually makes use of `packaging.requirements`. So "old style lazy loading" just means import at function runtime.